### ISSUE TITLE
Upgrade to 2.0

### ISSUE THREAD

**Issue by nasir-awan** [kerryjiang/SuperSocket#536]
we are using SuperSocker v1.6, and now we are upgrading to v2.0 that is .net core version.
In v1.6 we are using following nugets/namespaces

1. using SuperSocket.SocketBase;
2. using SuperSocket.SocketBase.Logging;
3. using SuperSocket.SocketBase.Protocol;
4. using SuperSocket.WebSocket.Config;
5. using SuperSocket.WebSocket.SubProtocol;

what are equivalent packages in v2.0 ? is there guideline to upgrade from v1.6 to v2.0?

Thanks,

**Comment by ScarletKuro** [comment]
I guess it would be something like this

1. SuperSocket.Primitives
2. No such thing, the logging is provided via Microsoft.Extensions.Logging
3. SuperSocket.ProtoBase
4 and 5. SuperSocket.WebSocket/SuperSocket.WebSocket.Server

You need a deep integration with Microsoft.Extensions.DependencyInjection.Abstractions and
Microsoft.Extensions.Hosting.Abstractions in order to use v2

**Comment by nasir-awan** [comment]
Thanks @ScarletKuro

**Comment by nasir-awan** [comment]
possible to use simple injector with v2?

**Comment by ScarletKuro** [comment]
Yes, I'm using in production `SimpleInjector` with `SupeSocket`(2.0).
But with little trick https://docs.simpleinjector.org/en/latest/generichostintegration.html
I cross wired the `SimpleInjector` and `Microsoft.DependencyInjection` so in fact I have two dependency injections which is fine.
Everything that was registered in Microsoft.DI can be resolved via SimpleInjector but not vice versa.
So most 3rd party that requires `Microsoft.Extensions.Hosting`(Orleans, SupeSocket, etc) I register in Microsoft.DI and my application layer I register in SimpleInjector(services, viewmodels, mappings etc).
